bokehjs/src/lib/models/tiles/tile_renderer.ts

```diff
@@ -55,10 +55,9 @@ export class TileRendererView extends RendererView {
     const y_start = y_range.start
     const x_end = x_range.end
     const y_end = y_range.end
-    assert(isFinite(x_start))
-    assert(isFinite(y_start))
-    assert(isFinite(x_end))
-    assert(isFinite(y_end))
+    if (!(isFinite(x_start) && isFinite(y_start) && isFinite(x_end) && isFinite(y_end))) {
+        logger.warn("tile extent is (partially) undefined")
```

Comment:
```suggestion
        logger.warn("tile extent is not fully defined")
```
IMO this slight change covers all the cases -- partially or fully undefined -- a little better. If you want to make this even more informative you could also include the four range start/end values in the message.